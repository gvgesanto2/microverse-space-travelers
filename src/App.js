import { Route, Routes } from 'react-router-dom';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';

import getRocketsFromApi from './redux/rocket/rocketApiAction';
import Navbar from './components/navbar/navbar.component';
import MissionsPage from './pages/missions-page/missions-page.component';
import MyProfile from './pages/my-profile/my-profile.component';
import RocketsPage from './pages/rockets-page/rockets-page.component';
import { fetchMissionsStartAsync } from './redux/mission/mission.actions';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissionsStartAsync());
    dispatch(getRocketsFromApi());
  }, []);

  return (
    <Routes>
      <Route path="/" element={<Navbar />}>
        <Route index element={<RocketsPage />} />
        <Route path="/missions" element={<MissionsPage />} />
        <Route path="/my-profile" element={<MyProfile />} />
      </Route>
    </Routes>
  );
}

export default App;
