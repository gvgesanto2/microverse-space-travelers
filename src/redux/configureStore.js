import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import rocketReducer from './rocket/rocket.reducer';
import getRocketsFromApi from './rocket/rocketApiAction';

// Initial Data
const preloadedState = {
  rockets: [],
};

// Establish the root reducer
const rootReducer = combineReducers({
  rockets: rocketReducer,
});

// Set up the Store
const ulisesStore = configureStore({
  reducer: rootReducer,
  preloadedState,
});

ulisesStore.dispatch(getRocketsFromApi());

export default ulisesStore;
