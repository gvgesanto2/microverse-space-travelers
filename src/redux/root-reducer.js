/* eslint-disable import/prefer-default-export */
import { combineReducers } from 'redux';
import missionReducer from './mission/mission.reducer';
import rocketReducer from './rocket/rocket.reducer';

export const rootReducer = combineReducers({
  rocket: rocketReducer,
  mission: missionReducer,
});
