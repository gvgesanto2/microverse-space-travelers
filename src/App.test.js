import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import App from './App';
import { store } from './redux/store';

const renderWithProviders = (reduxStore, route) => render(
  <Provider store={reduxStore}>
    <MemoryRouter initialEntries={[route]}>
      <App />
    </MemoryRouter>
  </Provider>,
);

describe('App component', () => {
  it('should render correctly with the Rockets Page content', () => {
    const component = renderWithProviders(store, '/');
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with the Missions Page content', () => {
    const component = renderWithProviders(store, '/missions');
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with the MyProfile Page content', () => {
    const component = renderWithProviders(store, '/my-profile');
    expect(component).toMatchSnapshot();
  });
});
