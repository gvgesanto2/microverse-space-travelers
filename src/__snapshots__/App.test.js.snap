// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly with the Missions Page content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="navbar"
      >
        <a
          class="navbar__logo"
          href="/"
        >
          <img
            alt="Space Travelers Logo"
            class="navbar__logo-img"
            src="logo.png"
          />
          <p
            class="navbar__logo-text"
          >
            Space Travelers' Hub
          </p>
        </a>
        <nav
          class="navbar__nav"
        >
          <ul
            class="navbar__nav-list"
          >
            <li>
              <a
                class="navbar__nav-link"
                href="/"
              >
                rockets
              </a>
            </li>
            <li>
              <a
                aria-current="page"
                class="navbar__nav-link navbar__nav-link--active"
                href="/missions"
              >
                missions
              </a>
            </li>
            <li>
              <a
                class="navbar__nav-link"
                href="/my-profile"
              >
                my profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <main
        class="main"
      >
        <div
          class="missions-page"
        >
          <table
            class="missions-table"
          >
            <thead>
              <tr>
                <th>
                  mission
                </th>
                <th>
                  description
                </th>
                <th>
                  status
                </th>
                <th>
                   
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="navbar"
    >
      <a
        class="navbar__logo"
        href="/"
      >
        <img
          alt="Space Travelers Logo"
          class="navbar__logo-img"
          src="logo.png"
        />
        <p
          class="navbar__logo-text"
        >
          Space Travelers' Hub
        </p>
      </a>
      <nav
        class="navbar__nav"
      >
        <ul
          class="navbar__nav-list"
        >
          <li>
            <a
              class="navbar__nav-link"
              href="/"
            >
              rockets
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="navbar__nav-link navbar__nav-link--active"
              href="/missions"
            >
              missions
            </a>
          </li>
          <li>
            <a
              class="navbar__nav-link"
              href="/my-profile"
            >
              my profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="main"
    >
      <div
        class="missions-page"
      >
        <table
          class="missions-table"
        >
          <thead>
            <tr>
              <th>
                mission
              </th>
              <th>
                description
              </th>
              <th>
                status
              </th>
              <th>
                 
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App component should render correctly with the MyProfile Page content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="navbar"
      >
        <a
          class="navbar__logo"
          href="/"
        >
          <img
            alt="Space Travelers Logo"
            class="navbar__logo-img"
            src="logo.png"
          />
          <p
            class="navbar__logo-text"
          >
            Space Travelers' Hub
          </p>
        </a>
        <nav
          class="navbar__nav"
        >
          <ul
            class="navbar__nav-list"
          >
            <li>
              <a
                class="navbar__nav-link"
                href="/"
              >
                rockets
              </a>
            </li>
            <li>
              <a
                class="navbar__nav-link"
                href="/missions"
              >
                missions
              </a>
            </li>
            <li>
              <a
                aria-current="page"
                class="navbar__nav-link navbar__nav-link--active"
                href="/my-profile"
              >
                my profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <main
        class="main"
      >
        <section
          class="my-profile"
        >
          <div
            class="missions-reserved"
          >
            <h2
              class="missions-reserved__title"
            >
              My Missions
            </h2>
            <ul
              class="missions-reserved__list"
            />
          </div>
          <section
            class="rocket-profile"
          >
            <h2>
              My Rockets
            </h2>
            <p>
              You have not reserved any rockets
            </p>
          </section>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="navbar"
    >
      <a
        class="navbar__logo"
        href="/"
      >
        <img
          alt="Space Travelers Logo"
          class="navbar__logo-img"
          src="logo.png"
        />
        <p
          class="navbar__logo-text"
        >
          Space Travelers' Hub
        </p>
      </a>
      <nav
        class="navbar__nav"
      >
        <ul
          class="navbar__nav-list"
        >
          <li>
            <a
              class="navbar__nav-link"
              href="/"
            >
              rockets
            </a>
          </li>
          <li>
            <a
              class="navbar__nav-link"
              href="/missions"
            >
              missions
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="navbar__nav-link navbar__nav-link--active"
              href="/my-profile"
            >
              my profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="main"
    >
      <section
        class="my-profile"
      >
        <div
          class="missions-reserved"
        >
          <h2
            class="missions-reserved__title"
          >
            My Missions
          </h2>
          <ul
            class="missions-reserved__list"
          />
        </div>
        <section
          class="rocket-profile"
        >
          <h2>
            My Rockets
          </h2>
          <p>
            You have not reserved any rockets
          </p>
        </section>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App component should render correctly with the Rockets Page content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="navbar"
      >
        <a
          aria-current="page"
          class="navbar__logo active"
          href="/"
        >
          <img
            alt="Space Travelers Logo"
            class="navbar__logo-img"
            src="logo.png"
          />
          <p
            class="navbar__logo-text"
          >
            Space Travelers' Hub
          </p>
        </a>
        <nav
          class="navbar__nav"
        >
          <ul
            class="navbar__nav-list"
          >
            <li>
              <a
                aria-current="page"
                class="navbar__nav-link navbar__nav-link--active"
                href="/"
              >
                rockets
              </a>
            </li>
            <li>
              <a
                class="navbar__nav-link"
                href="/missions"
              >
                missions
              </a>
            </li>
            <li>
              <a
                class="navbar__nav-link"
                href="/my-profile"
              >
                my profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <main
        class="main"
      >
        <div>
          <section
            class="rocket-list-container"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="navbar"
    >
      <a
        aria-current="page"
        class="navbar__logo active"
        href="/"
      >
        <img
          alt="Space Travelers Logo"
          class="navbar__logo-img"
          src="logo.png"
        />
        <p
          class="navbar__logo-text"
        >
          Space Travelers' Hub
        </p>
      </a>
      <nav
        class="navbar__nav"
      >
        <ul
          class="navbar__nav-list"
        >
          <li>
            <a
              aria-current="page"
              class="navbar__nav-link navbar__nav-link--active"
              href="/"
            >
              rockets
            </a>
          </li>
          <li>
            <a
              class="navbar__nav-link"
              href="/missions"
            >
              missions
            </a>
          </li>
          <li>
            <a
              class="navbar__nav-link"
              href="/my-profile"
            >
              my profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="main"
    >
      <div>
        <section
          class="rocket-list-container"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
